### PR TITLE
Update JDK Quickstarters

### PR DESCRIPTION
**Pull request by braisvq1996**
Related to https://github.com/opendevstack/ods-quickstarters/issues/966

**JDK agent**
- [x] Build successfull with java 11, 17 and 21
- [x] Switch between java versions works

**SpringBoot**
- [x] Provision successfull
- [x] Build successfull
- [x] Example app working

**Spock**
- [x] Provision successfull
- [x] Build successfull
- [x] Example app working

### CODE REVIEW

**Comment by renedupont** [comment]
The build failed

**Comment by braisvq1996** [comment]
> The build failed 

The mirror from terrarium jdk failed last build. Now it is working

**Review by renedupont — APPROVED** [review]

File: e2e-spock-geb/Jenkinsfile.template [hunk 1]
```diff
@@ -24,7 +24,7 @@ def stageTest(def context) {
   stage('Integration Test') {
   	sh (script: "chmod a+x gradle*", label : "allow gradle to execute")
     withEnv(["TAGVERSION=${context.tagversion}", "NEXUS_HOST=${context.nexusHost}", "NEXUS_USERNAME=${context.nexusUsername}", "NEXUS_PASSWORD=${context.nexusPassword}", "JAVA_OPTS=${javaOpts}","GRADLE_TEST_OPTS=${gradleTestOpts}","ENVIRONMENT=${springBootEnv}","OPENSHIFT_PROJECT=${context.targetProject}","OPENSHIFT_APP_DOMAIN=${context.getOpenshiftApplicationDomain()}"]) {
-      def status = sh(script: "./gradlew clean test --stacktrace --no-daemon", returnStatus: true)
+      def status = sh(script: "source use-j17.sh && ./gradlew clean test --stacktrace --no-daemon && source use-j11.sh", returnStatus: true)
```
[the remark this answers is not in the document]
Comment (braisvq1996) [reply e2e-spock-geb/Jenkinsfile.template:27]: I did tried before updating it and it gives an error when using java 21. That is why I made it use java 17 instead

`````
java.lang.IllegalArgumentException: Unsupported class file major version 65
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:199)
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:180)
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:166)
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:287)
	at org.codehaus.groovy.ast.decompiled.AsmDecompiler.parseClass(AsmDecompiler.java:83)
	at org.codehaus.groovy.control.ClassNodeResolver.findDecompiled(ClassNodeResolver.java:255)
	at org.codehaus.groovy.control.ClassNodeResolver.tryAsLoaderClassOrScript(ClassNodeResolver.java:193)
	at org.codehaus.groovy.control.ClassNodeResolver.findClassNode(ClassNodeResolver.java:175)
	at org.codehaus.groovy.control.ClassNodeResolver.resolveName(ClassNodeResolver.java:129)
	at org.codehaus.groovy.ast.decompiled.AsmReferenceResolver.resolveClassNullable(AsmReferenceResolver.java:57)
	at org.codehaus.groovy.ast.decompiled.AsmReferenceResolver.resolveClass(AsmReferenceResolver.java:44)
	at org.codehaus.groovy.ast.decompiled.ClassSignatureParser.configureClass(ClassSignatureParser.java:49)
	at org.codehaus.groovy.ast.decompiled.DecompiledClassNode.lazyInitSupers(DecompiledClassNode.java:189)
	at org.codehaus.groovy.ast.decompiled.DecompiledClassNode.getAnnotations(DecompiledClassNode.java:136)
	at org.codehaus.groovy.ast.ClassNode.getAnnotations(ClassNode.java:1540)
	at org.spockframework.compiler.AstUtil.getAnnotation(AstUtil.java:62)
	at org.spockframework.compiler.AstUtil.hasAnnotation(AstUtil.java:54)
	at org.spockframework.compiler.SpecAnnotator.isRepeatableExtensionAnnotation(SpecAnnotator.java:181)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:178)
	at java.base/java.util.stream.Streams$StreamBuilderImpl.forEachRemaining(Streams.java:411)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1708)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
	at org.spockframework.compiler.SpecAnnotator.addRepeatedExtensionAnnotations(SpecAnnotator.java:98)
	at org.spockframework.compiler.SpecAnnotator.visitSpec(SpecAnnotator.java:55)
	at org.spockframework.compiler.model.Spec.accept(Spec.java:110)
	at org.spockframework.compiler.SpockTransform$Impl.processSpec(SpockTransform.java:77)
	at org.spockframework.compiler.SpockTransform$Impl.visit(SpockTransform.java:63)
	at org.spockframework.compiler.SpockTransform.visit(SpockTransform.java:48)
	at org.codehaus.groovy.transform.ASTTransformationVisitor.lambda$addPhaseOperationsForGlobalTransforms$5(ASTTransformationVisitor.java:377)
	at org.codehaus.groovy.control.CompilationUnit$ISourceUnitOperation.doPhaseOperation(CompilationUnit.java:896)
	at org.codehaus.groovy.control.CompilationUnit.processPhaseOperations(CompilationUnit.java:692)
	at org.codehaus.groovy.control.CompilationUnit.compile(CompilationUnit.java:666)
	at org.codehaus.groovy.control.CompilationUnit.compile(CompilationUnit.java:647)
	at org.gradle.api.internal.tasks.compile.ApiGroovyCompiler.execute(ApiGroovyCompiler.java:285)
	at org.gradle.api.internal.tasks.compile.ApiGroovyCompiler.execute(ApiGroovyCompiler.java:67)
	at org.gradle.api.internal.tasks.compile.GroovyCompilerFactory$DaemonSideCompiler.execute(GroovyCompilerFactory.java:90)
	at org.gradle.api.internal.tasks.compile.GroovyCompilerFactory$DaemonSideCompiler.execute(GroovyCompilerFactory.java:76)
	at org.gradle.api.internal.tasks.compile.daemon.AbstractIsolatedCompilerWorkerExecutor$CompilerWorkAction.execute(AbstractIsolatedCompilerWorkerExecutor.java:78)
	at org.gradle.workers.internal.DefaultWorkerServer.execute(DefaultWorkerServer.java:63)
	at org.gradle.workers.internal.AbstractClassLoaderWorker$1.create(AbstractClassLoaderWorker.java:54)
	at org.gradle.workers.internal.AbstractClassLoaderWorker$1.create(AbstractClassLoaderWorker.java:48)
	at org.gradle.internal.classloader.ClassLoaderUtils.executeInClassloader(ClassLoaderUtils.java:100)
	at org.gradle.workers.internal.AbstractClassLoaderWorker.executeInClassLoader(AbstractClassLoaderWorker.java:48)
	at org.gradle.workers.internal.IsolatedClassloaderWorker.run(IsolatedClassloaderWorker.java:49)
	at org.gradle.workers.internal.IsolatedClassloaderWorker.run(IsolatedClassloaderWorker.java:30)
	at org.gradle.workers.internal.WorkerDaemonServer.run(WorkerDaemonServer.java:96)
	at org.gradle.workers.internal.WorkerDaemonServer.run(WorkerDaemonServer.java:65)
	at org.gradle.process.internal.worker.request.WorkerAction$1.call(WorkerAction.java:138)
	at org.gradle.process.internal.worker.child.WorkerLogEventListener.withWorkerLoggingProtocol(WorkerLogEventListener.java:41)
	at org.gradle.process.internal.worker.request.WorkerAction.lambda$run$0(WorkerAction.java:135)
	at org.gradle.internal.operations.CurrentBuildOperationRef.with(CurrentBuildOperationRef.java:80)
	at org.gradle.process.internal.worker.request.WorkerAction.run(WorkerAction.java:127)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:414)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.AbstractManagedExecutor$1.run(AbstractManagedExecutor.java:47)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
````

[the remark this answers is not in the document]
Comment (stitakis) [reply e2e-spock-geb/Jenkinsfile.template:27]: from the stack trace looks to me that the library asm does not support Java 21... which version of groovy is being used? version 4?

[the remark this answers is not in the document]
Comment (braisvq1996) [reply e2e-spock-geb/Jenkinsfile.template:27]: The version used is the latest available I could find, Spock 2.3-groovy-4.0

[the remark this answers is not in the document]
Comment (stitakis) [reply e2e-spock-geb/Jenkinsfile.template:27]: let us keep spock using java 17 for the time being... probably next version of groovy will support java 21

**Review by oalyman — APPROVED** [review]

**Comment by braisvq1996** [comment]
> LGTM.
> 
> I think it would be nice to create a jdk 21 only agent as well. For teams that did the switch and would not need to have a build agent with 3 JVMs installed.

If we want to use SonarQube we still need java 11. Even if we update SonarQube to the version 9.9 LTS the scanner will still needs java 17 so it is not possible to have an agent with only 1 JDK regrettably

**Pull request merged**